### PR TITLE
ENH New method to get metadata without a network request

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/11428

Manually tag 1.1.0 after merging

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,11 +3,11 @@
 Metadata and some supporting PHP logic for determining which branches of various GitHub repositories relate to which versions of Silverstripe CMS.
 
 > [!IMPORTANT]
-> Only the `main` branch of this repository is maintained.
+> Only the `main` branch of this repository is maintained. Stable tags must be manually created from the `main` branch. There is no auto tagging via a GitHub action. The `3`, `4`, and `5` branches are legacy and should not be referenced going forward.
```
Comment (GuySartorelli) [inline README.md:6]:
```suggestion
> Only the `main` branch of this repository is maintained. Stable tags must be manually created from the `main` branch. There is no auto tagging via a GitHub action. All branches other than `main` are legacy and should not be referenced going forward.
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (emteknetnz) [reply README.md]: Updated

File: src/MetaData.php [hunk 2]
```diff
@@ -179,13 +179,30 @@ public static function removeReposNotInCmsMajor(array $metadata, string|int $cms
...
+     * Use this if making a network request is not possible or desired, and the downside from the data
+     * being potentially out-of-date is acceptable.
+     * @param bool $categorised If true, output is grouped by category.
+     */
+    public static function getAllRepositoryMetaDataFromLocal(bool $categorised = true): array
+    {
+        return MetaData::getRepositoryMetaDataInner($categorised, true);
+    }
+
+    private static function getRepositoryMetaDataInner(bool $categorised = true, bool $doFetch = true): array
```
Comment (GuySartorelli) [inline src/MetaData.php:202]: Please just add a new optional param to the existing method instead. This repo isn't protected by our supported modules policies and there's 0 chance someone is subclassing this so it isn't a breaking change.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/MetaData.php (hunk 2 above)
Comment (emteknetnz) [reply src/MetaData.php]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**